File: LeetCode75-sols/Array_String/Java/Product_of_Array_Except_Self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] answer = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            answer[i] = 1;
        }
        int prodval = 1;
        // Create left products
        for (int i = 1; i < nums.length; i++) {
            answer[i] = answer[i-1] * nums[i-1];
        }
        // Create right products and merge
        for (int i = nums.length - 1; i >= 0; i--) {
            answer[i] *= prodval;
            prodval *= nums[i];
        }
        return answer;
    }
}

File: Top_Interview_150/Array_String/Python/Product_of_Array_Except_Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        answer = [1] * len(nums)
        for i in range(1, len(nums)):
            answer[i] = answer[i - 1] * nums[i - 1]
        suffix_product = 1
        for i in range(len(nums) - 1, -1, -1):
            answer[i] *= suffix_product
            suffix_product *= nums[i]
        return answer
